### ISSUE TITLE
Problem detecting connectivity problems using enableKeepAlive method

### ISSUE THREAD

**Issue by kai670** [legastero/stanza#401]
I'm using a stanza client with the keepalive enable with the next options:

```
client.enableKeepAlive({interval: 20, timeout: 10});
```

But when the connection is lost, it is not working as I would expect. What I'm seeing is:

```
  0 -> Keepalive sent and answer received
  5 -> Connection is lost
 20 -> Keepalive sent
 30 -> Timeout waiting for server answer
101 -> A new connection is established resuming previous XML stream
```

Where the first number is the number of seconds elapsed since the start.

I was expecting to see the session reestablished right after the timeout waiting for the server answer but I have to wait 61 additional seconds and that is a lost of time...

**Comment by kai670** [comment]
I was investigating a bit about this and I have some new data:

First of all, I previously forgot reconnection is handled by the application and stanza just informs that the socket was broken with a "disconnected" event. Besides, a 10s delay was being introduced by the application code.

Everything works as expected in stanza about ping interval and timeout time. The unexpected delay is caused because stanza doesn't emit the "disconnected" event until the WS is really disconnected. After the configured timeout, stanza just requests the WS to be closed. 60 seconds after that the WS is finally closed with a 1006 reason code and then stanza emits the "disconnected" event.

Additionally, when the first timeout occurs the keepalive process is not stopped and keeps sending pings to the server.

**Comment by kai670** [comment]
This is a summary of the events I see happening after a connection lost:

* 14:36:42
    * PING sent by stanza
* 14:36:52
    * Timeout after waiting 10s for server response
    * Emits 'stream:error' event and calls `client.transport.disconnect();`
* 14:37:02
  * Another PING is sent by stanza. (afaik this should not be happening as a previous one timeouted)
* 14:37:12
    * Timeout of the previous ping
* More pings and timeout occurs here...
* 14:37:53
  * Websocket state is set to close, `onClose` handler is run and stanza emits the "disconnected" events
* "disconnected" events keep being sent...

So summing up, I see two main problems:
 - Keep alive process is not stopped and subsequent wrong disconnected events are emitted.
 - Applications have to wait for 60 additional seconds before being able to know connections is lost and try to reconnect. That seems a huge delay to me.
